src/lib.rs

```diff
@@ -250,72 +323,149 @@ impl UinputAbsSetup {
 ///
 /// The meaning of the "code" and "value" fields will depend on the underlying type of event.
 #[derive(Copy, Clone)]
-#[repr(transparent)]
-pub struct InputEvent(input_event);
-
-impl InputEvent {
-    /// Returns the timestamp associated with the event.
-    #[inline]
-    pub fn timestamp(&self) -> SystemTime {
-        timeval_to_systime(&self.0.time)
-    }
-
-    /// Returns the type of event this describes, e.g. Key, Switch, etc.
-    #[inline]
-    pub fn event_type(&self) -> EventType {
-        EventType(self.0.type_)
-    }
+pub enum InputEvent {
+    /// [`SynchronizationEvent`]
+    Synchronization(SynchronizationEvent),
+    /// [`KeyEvent`]
```

Comment:
<img width="258" alt="Screenshot 2023-03-19 at 00 18 52" src="https://user-images.githubusercontent.com/704250/226153393-2bf5a8ca-ddc8-4224-bfd9-235b136043ba.png">

i truly admire the hustle and dedication to documentation here, but in this case, as demonstrated in the screenshot, the doc comments are extraneous! what do you think, should we leave them in the docs for now or should we figure out something to say on the rustdoc page for this enum?

Reply:
You are right, this is a bit redundant. I guess we should remove the doc comments.